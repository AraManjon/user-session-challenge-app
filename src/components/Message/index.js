import React from 'react';
import PropTypes from 'prop-types';
var moment = require('moment');

const Message = ({ session }) => {
    return (
        <div className="writer">
            <p className="parraf" > Hello {session.username} <span role="img" aria-label="hello">ðŸ‘‹</span> </p>
            <p className="parraf" >This is your session key </p>
            <p className="parraf__text-red" >{session.token}</p>
            <p className="parraf" >{moment(session.date).format('DD-MM-YYYY THH:mm:ss')}</p>
        </div>
	);
};
Message.propTypes = {
    session: PropTypes.object
}
export default Message;