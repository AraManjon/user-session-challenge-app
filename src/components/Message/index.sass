.parraf
    word-wrap: break-word
    text-align: center
    margin: 0.5rem 0
    color: $black
    line-height: 2.2rem
.parraf
    &__text-red
        color: $red
        word-wrap: break-word
        text-align: center
        margin: 0.5rem 0
        line-height: 2.2rem