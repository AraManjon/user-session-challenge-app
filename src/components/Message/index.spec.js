import React from "react";
import { create } from "react-test-renderer";
import Message from "./index";

describe("message", () => {
  it("show a message with session information when session exist", () => {

    const date = new Date(2020, 4, 16, 21, 18, 36);
    const session = {
      username: "myName",
      token: "token",
      date: date,
    };

    jest.mock('moment', () => () => ({format: () => '16-05-2020 T21:18:36'}));

    const message = create(<Message session={session} />);
    const messageResult = message.toJSON();

    expect(messageResult).toMatchSnapshot();
  });
});
