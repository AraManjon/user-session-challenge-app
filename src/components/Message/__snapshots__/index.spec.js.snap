// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message show a message with session information when session exist 1`] = `
<div
  className="writer"
>
  <p
    className="parraf"
  >
     Hello 
    myName
     
    <span
      aria-label="hello"
      role="img"
    >
      ðŸ‘‹
    </span>
     
  </p>
  <p
    className="parraf"
  >
    This is your session key 
  </p>
  <p
    className="parraf__text-red"
  >
    token
  </p>
  <p
    className="parraf"
  >
    16-05-2020 T21:18:36
  </p>
</div>
`;
